lib/foreman/renderer/scope/macros/host_template.rb

```diff
@@ -121,6 +121,41 @@ def ks_console
             (@port && @baud) ? "console=ttyS#{@port},#{@baud}" : ''
           end
 
+          apipie :method, 'Generates commands for package(s) installation, supported OS families: Redhat, Debian & Suse' do
+            required :packages, String, desc: "Package(s) to install"
+            returns String, desc: 'Installation commands'
+            example "CentOS host: install_packages('pkg1') #=> yum -y install pkg1"
+            example "Fedora host: install_packages('pkg1 pkg2') #=> dnf -y install pkg1 pkg2"
+          end
+          def install_packages(packages)
+            raise UndefinedPackages.new if packages.empty?
+            raise UnsupportedOS.new unless host.operatingsystem && ['Redhat', 'Debian', 'Suse'].include?(host.operatingsystem.family)
+
+            if host.operatingsystem.family == 'Redhat'
+              if host.operatingsystem.name.downcase == 'fedora'
+                return "dnf -y install #{packages}"
+              else
+                return "yum -y install #{packages}"
+              end
+            end
+
+            if host.operatingsystem.family == 'Debian'
+              return <<~CMD
+                [ -x "$(command -v subscription-manager)" ] && subscription-manager refresh
+                export DEBIAN_FRONTEND=noninteractive
+                apt-get -y update
+                apt-get -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" -y install #{packages}
+              CMD
+            end
+
+            if host.operatingsystem.family == 'Suse'
+              <<~CMD
+                zypper refresh
+                zypper -n install #{packages}
+              CMD
+            end
```

Comment:
Mind making this a `case host.operatingsystem.family` statement instead of multiple if equal statements?